### DISCUSSION TITLE
NordVPN Wireguard not working properly

### DISCUSSION THREAD

**Discussion by rjuarezp** [qdm12/gluetun#1911] [Q&A] [answered]
Hi!
I have setup a container following the docker-cli from this website [https://github.com/qdm12/gluetun-wiki/blob/main/setup/providers/nordvpn.md](https://github.com/qdm12/gluetun-wiki/blob/main/setup/providers/nordvpn.md) and added NET_ADM as container capability.

I got my token key from the nordvpn website and put it as value of the WIREGUARD_PRIVATE_KEY environmental variable.

Now the container is trying to contact NordVPN but I get the following error:

`========================================

========================================

=============== gluetun ================

========================================

=========== Made with ❤️ by ============

======= https://github.com/qdm12 =======

========================================

========================================

Running version latest built on 2023-10-07T13:26:08.155Z (commit 1c43a1d)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new

🐛 Bug? https://github.com/qdm12/gluetun/issues/new

✨ New feature? https://github.com/qdm12/gluetun/issues/new

☕ Discussion? https://github.com/qdm12/gluetun/discussions/new

💻 Email? quentin.mcgaw@gmail.com

💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12

2023-10-15T15:41:58Z INFO [routing] default route found: interface eth0, gateway 172.17.0.1, assigned IP 172.17.0.11 and family v4

2023-10-15T15:41:58Z INFO [routing] local ethernet link found: eth0

2023-10-15T15:41:58Z INFO [routing] local ipnet found: 172.17.0.0/16

2023-10-15T15:41:58Z INFO [firewall] enabling...

2023-10-15T15:41:58Z INFO [firewall] enabled successfully

2023-10-15T15:42:00Z INFO [storage] merging by most recent 17689 hardcoded servers and 17689 servers read from /gluetun/servers.json

2023-10-15T15:42:00Z ERROR VPN settings: Wireguard settings: private key is not valid: wgtypes: incorrect key size: 48

`

I have copied the code completely and I am sure that there is no mistake there. Does anybody know what is happening here?

Thanks!

**Comment by bnhf** [comment]
> I have setup a container following the docker-cli from this website https://github.com/qdm12/gluetun-wiki/blob/main/setup/providers/nordvpn.md and added NET_ADM as container capability.

Should be ```NET_ADMIN``` not ```NET_ADM```

**Reply by rjuarezp** [reply]
Sorry! Typo... It was NET_ADMIN... otherwise iptables firewall enabling would not work. It was my mistake. This is not the problem... But thanks!!! By the way... connecting with the openvpn protocol works without problems...

**Comment by bnhf** [comment, marked as answer]
> Wireguard settings: private key is not valid: wgtypes: incorrect key size: 48

Your Wireguard Private Key for NordVPN should be 44 characters not 48. There's a process you need to go through to extract the Private Key from the token -- have you done that?

**Reply by rjuarezp** [reply]
Which process? Maybe that's the problem... My token is 64 characters long... How do you get the key from the token?

**Reply by bnhf** [reply]
Follow the links in the Wiki about obtaining your Wireguard Private Key. I think it's been something of a moving target with NordVPN. Last time I did it, I used the https://github.com/bubuntux/nordvpn/pkgs/container/nordvpn/83974496?tag=get_private_key method, but I'm not sure that works anymore.

**Reply by rjuarezp** [reply]
I found the website... [https://gist.github.com/bluewalk/7b3db071c488c82c604baf76a42eaad3](https://gist.github.com/bluewalk/7b3db071c488c82c604baf76a42eaad3)! Thanks! That was it!!!
